src/Components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,162 @@
+import classNames from 'classnames';
+import { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onDelete: (value: number) => void;
+  isIncludes: boolean;
+  onChangeStatusTodo?: (todoID: number) => void;
+  onChangeNewTitle?: (todoID: number, todoNewTitle: string) => void;
+  setErrorMessage: (errorMessage: string) => void;
+};
+
+export const TodoInfo: React.FC<Props> = ({
+  todo,
+  onDelete,
+  isIncludes,
+  onChangeStatusTodo,
+  onChangeNewTitle,
+  setErrorMessage,
+}) => {
+  const { title, id } = todo;
+
+  const [isClicked, setIsClicked] = useState(false);
+  const [todoTitle, setTodoTitle] = useState(todo.title);
+  const [isChangedTitle, setIsChangedTitle] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleChangeTitleTodo = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const {
+      target: { value },
+    } = event;
+
+    setTodoTitle(value);
+  };
+
+  const handleUpdateTittleTodo = async (
+    event: React.ChangeEvent<HTMLFormElement>,
+  ) => {
+    event.preventDefault();
+
+    setIsClicked(false);
+
+    if (todoTitle !== todo.title) {
+      try {
+        setIsChangedTitle(true);
+        await onChangeNewTitle?.(id, todoTitle);
+      } catch {
+        setErrorMessage('Unable to edit a todo');
+      } finally {
+        setIsChangedTitle(false);
+      }
+    }
+
+    return '';
```

Comment:
There is no need to return an empty string in the 'handleUpdateTittleTodo' function. You can simply remove this line.